src/Client.php

```diff
@@ -57,33 +55,34 @@ function (RequestInterface $request) {
         $this->authenticator = $authenticatorFactory->getAuthenticator($clientFactory, 'https://vault.azure.net');
     }
 
-    private function sendRequest(Request $request)
+    private function sendRequest(Request $request): array
     {
         try {
-            if (!$this->token) {
+            if (empty($this->token)) {
                 $this->token = $this->authenticator->getAuthenticationToken();
                 $this->logger->info('Successfully authenticated.');
             }
             $response = $this->guzzle->send($request);
-            return \GuzzleHttp\json_decode($response->getBody()->getContents(), true);
-        } catch (GuzzleException $e) {
+            return (array) json_decode($response->getBody()->getContents(), true, 512, JSON_THROW_ON_ERROR);
+        } catch (JsonException | GuzzleException $e) {
```

Comment:
`handleRequestException` ted sice prijima `Throwable`, ale fakticky neco dela jen pri `RequestException` nejak reaguje. Spis bych tu tak volil cestu aby `handleRequestException` akceptoval `RequestException` a sem hodil `if` na to, jestli `$e` je `RequestException`.

Reply:
refs/tags/dev-odin-PS-3733